services/motion/builtin/move_request.go

```diff
@@ -177,33 +181,158 @@ func (ms *builtIn) newMoveOnGlobeRequest(
+				return nil, errors.New("could not interpret motion_profile field as string")
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+		if costFactorRaw, ok := extra["replan_cost_factor"]; ok {
+			costFactor, ok := costFactorRaw.(float64)
+			if !ok {
+				return nil, errors.New("could not interpret replan_cost_factor field as float")
+			}
+			replanCostFactor = costFactor
+		}
+	}
+
 	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, localizer, limits, kinematicsOptions)
 	if err != nil {
 		return nil, err
 	}
 
+	geomsRaw := spatialmath.GeoObstaclesToGeometries(obstacles, origin)
+
+	mr, err := relativeMoveRequestFromAbsolute(
+		ctx,
+		motionCfg,
+		ms.logger,
+		kb,
+		goalPoseRaw,
+		fs,
+		geomsRaw,
+		extra,
+	)
+	if err != nil {
+		return nil, err
+	}
+	mr.seedPlan = seedPlan
+	mr.replanCostFactor = replanCostFactor
+	return mr, nil
+}
+
+// newMoveOnMapRequest instantiates a moveRequest intended to be used in the context of a MoveOnMap call.
+func (ms *builtIn) newMoveOnMapRequest(
+	ctx context.Context,
+	componentName resource.Name,
+	goalPoseRaw spatialmath.Pose,
+	slamName resource.Name,
+	kinematicsOptions kinematicbase.Options,
+	extra map[string]interface{},
+) (*moveRequest, error) {
+	// get the SLAM Service from the slamName
+	slamSvc, ok := ms.slamServices[slamName]
+	if !ok {
+		return nil, resource.DependencyNotFoundError(slamName)
+	}
+
+	// gets the extents of the SLAM map
+	limits, err := slam.Limits(ctx, slamSvc)
+	if err != nil {
+		return nil, err
+	}
+	limits = append(limits, referenceframe.Limit{Min: -2 * math.Pi, Max: 2 * math.Pi})
+
+	// create a KinematicBase from the componentName
+	component, ok := ms.components[componentName]
+	if !ok {
+		return nil, resource.DependencyNotFoundError(componentName)
+	}
+	b, ok := component.(base.Base)
+	if !ok {
+		return nil, fmt.Errorf("cannot move component of type %T because it is not a Base", component)
+	}
+
+	if extra != nil {
+		if profile, ok := extra["motion_profile"]; ok {
+			motionProfile, ok := profile.(string)
+			if !ok {
+				return nil, errors.New("could not interpret motion_profile field as string")
+			}
+			kinematicsOptions.PositionOnlyMode = motionProfile == motionplan.PositionOnlyMotionProfile
+		}
+	}
+
+	fs, err := ms.fsService.FrameSystem(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, motion.NewSLAMLocalizer(slamSvc), limits, kinematicsOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// get point cloud data in the form of bytes from pcd
+	pointCloudData, err := slam.PointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return nil, err
+	}
+	// store slam point cloud data  in the form of a recursive octree for collision checking
+	octree, err := pointcloud.ReadPCDToBasicOctree(bytes.NewReader(pointCloudData))
+	if err != nil {
+		return nil, err
+	}
+
+	mr, err := relativeMoveRequestFromAbsolute(
```

Comment:
I think its mostly been copy pasted up until this point, correct?  Want to make sure I'm not missing anything

Reply:
Yeah everything above should be identical to what was in `planMoveOnMap` before. Some things got chopped out and/or moved to `relativeMoveRequest` but I don't think anything above this line is new.